### ISSUE TITLE
R6: Hard-zero particles in model forward pass

### ISSUE THREAD

**Issue by mateon01** [sa-project-private-repo/opensearch-neural-pre-train#26]
## Problem

Even with improved loss functions, the model may produce small residual activations for particle tokens due to ReLU output.

## Solution

Apply hard-zero mask in SPLADEDocContextGated.forward() post-ReLU:
```python
sparse_repr = torch.relu(logits)
if self.particle_mask is not None:
    sparse_repr = sparse_repr * self.particle_mask
```

## Priority
Low - Last resort if R1-R5 don't sufficiently suppress particles. May hurt retrieval if some particles carry semantic signal.

## Related
- src/model/splade_xlmr.py
